enhancements/network/on-prem-mutable-vips.md

```diff
@@ -0,0 +1,213 @@
+---
+title: on-prem-mutable-vips
+authors:
+  - "@mkowalski"
+reviewers:
+  - "@JoelSpeed, to review API"
+  - "@sinnykumari, to review MCO"
+  - "@cybertron, to peer-review OPNET"
+  - "@cgwalters"
+approvers:
+  - "@JoelSpeed"
+api-approvers:
+  - "@danwinship"
+  - "@JoelSpeed"
+creation-date: 2023-08-28
+last-updated: 2023-08-28
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-178
+  - https://issues.redhat.com/browse/OPNET-340
+  - https://issues.redhat.com/browse/OPNET-80
+see-also:
+  - "/enhancements/on-prem-dual-stack-vips.md"
+replaces:
+superseded-by:
+---
+
+# On-Prem Mutable VIPs
+
+## Summary
+
+Originally the on-prem loadbalancer architecture supported only single-stack IPv4 or IPv6 deployments. Later on, after dual-stack support was added to the on-prem deployments, a work has been done to allow installing clusters with a pair of virtual IPs (one per IP stack).
+This work however only covered installation-time, leaving clusters originally installed as single-stack and later converted to dual-stack out of scope.
+
+This design is proposing a change which will allow adding a second pair of virtual IPs to clusters that became dual-stack only during their lifetime.
+
+## Motivation
+
+We have customers who installed clusters before dual-stack VIP feature existed. Currently they have no way of migrating because the new feature is avaialble only during initial installation of the cluster.
+
+### User Stories
+
+* As a deployer of a dual-stack OpenShift cluster, I want to access the API using both IPv4 and IPv6.
+
+* As a deployer of a dual-stack OpenShift cluster, I want to access the Ingress using both IPv4 and IPv6.
+
+### Goals
+
+* Allow adding a second pair of virtual IPs to an already installed dual-stack cluster.
+
+* Allow deleting a second pair of virtual IPs on an already installed dual-stack cluster.
+
+### Non-Goals
+
+* Modifying existing virtual IP configuration.
+
+* Modifying virtual IP configuration after second pair of VIPs has been added. We only want to cover "create" and "delete" operation and only for the second pair of VIPs.
+
+* Configuration of any VIPs beyond a second pair for the second IP stack. MetalLB is a better solution for creating arbitrary loadbalancers.
+
+## Proposal
+
+### Workflow Description
```

Comment:
I know the following might vary for each platform, but is there any expectation that the user will open the traffic for the required ports for ingress and api?